homeassistant/components/proximity/coordinator.py

```diff
@@ -93,13 +95,52 @@ def __init__(
 
         self.state_change_data: StateChangedData | None = None
 
+    @callback
+    def async_add_entity_mapping(self, tracked_entity_id: str, entity_id: str) -> None:
+        """Add an tracked entity to proximity entity mapping."""
+        if tracked_entity_id not in self.entity_mapping:
+            self.entity_mapping[tracked_entity_id] = []
+        self.entity_mapping[tracked_entity_id].append(entity_id)
+
     async def async_check_proximity_state_change(
         self, entity: str, old_state: State | None, new_state: State | None
     ) -> None:
         """Fetch and process state change event."""
         self.state_change_data = StateChangedData(entity, old_state, new_state)
         await self.async_refresh()
 
+    async def async_check_tracked_entity_change(
+        self, event: EventType[er.EventEntityRegistryUpdatedData]
+    ) -> None:
+        """Fetch and process tracked entity change event."""
+        data = event.data
+        if data["action"] == "remove":
+            self._create_removed_tracked_entity_issue(data["entity_id"])
+
+        if data["action"] == "update" and "entity_id" in data["changes"]:
+            old_tracked_entity_id = data["old_entity_id"]
+            new_tracked_entity_id = data["entity_id"]
+
+            entity_reg = er.async_get(self.hass)
+            for rel_ent_id in self.entity_mapping.get(old_tracked_entity_id, []):
```

Comment:
Please use full words for the variable names. I don't understand `rel_ent_id`.

Reply:
1e7bac775b2a4a96f673e944df7e110b39eef0ad